gui/dewar_tree.py

```diff
@@ -89,173 +97,181 @@ def keyPressEvent(self, event):
-            if mountedIndex != None:
-                self.model.itemFromIndex(mountedIndex).setForeground(
-                    QtGui.QColor("red")
-                )
-                font = QtGui.QFont()
-                font.setUnderline(True)
-                font.setItalic(True)
-                font.setOverline(True)
-                self.model.itemFromIndex(mountedIndex).setFont(font)
-            self.parent.row_clicked(selectedSampleIndex)
-        elif selectedSampleIndex == None and collectionRunning == False:
-            if mountedIndex != None:
-                self.setCurrentIndex(mountedIndex)
-                self.model.itemFromIndex(mountedIndex).setForeground(
-                    QtGui.QColor("red")
-                )
-                font = QtGui.QFont()
-                font.setUnderline(True)
-                font.setItalic(True)
-                font.setOverline(True)
-                self.model.itemFromIndex(mountedIndex).setFont(font)
-                self.parent.row_clicked(mountedIndex)
-        else:
-            pass
-        if selectedIndex != None and collectionRunning == False:
-            self.setCurrentIndex(selectedIndex)
-            self.parent.row_clicked(selectedIndex)
-        if collectionRunning == True:
-            if mountedIndex != None:
-                self.setCurrentIndex(mountedIndex)
+        self.model.itemChanged.connect(self.queueSelectedSample)
         if self.isExpanded:
             self.expandAll()
         else:
             self.collapseAll()
         self.scrollTo(self.currentIndex(), QtWidgets.QAbstractItemView.PositionAtCenter)
 
+    def add_samples_to_puck_tree(self, puckContents, parentItem: QtGui.QStandardItem, index_label):
+        # Method will attempt to add samples to the puck. If you don't belong to the proposal,
+        # it will not add samples and clear the puck information
+        selectedIndex = None
+        mountedIndex = None
+        selectedSampleIndex = None
+        collectionRunning = False
+        for j, sample_id in enumerate(puckContents):
+            if not sample_id:
+                # this is an empty spot, no sample
+                position_s = str(j + 1)
+                item = QtGui.QStandardItem(QtGui.QIcon(ICON), position_s)
+                item.setData("", 32)
+                parentItem.appendRow(item)
+                continue
+
+            sample = sampleDict.get(
+                sample_id,
+                sampleDict.setdefault(sample_id, db_lib.getSampleByID(sample_id)),
+            )
+
+            #if not IS_STAFF and not self.is_proposal_member(sample["proposalID"]):
+            if not self.is_proposal_member(sample["proposalID"]):
+                # If the user is not part of the proposal or is not staff, don't fill tree
+                # Clear the puck information and don't make it selectable
+                # parentItem.clearData()
+                parentItem.setText(index_label)
```

Comment:
a question - should this be made more visibly occupied but not available to the current user? it wouldn't surprise me if, for example, beamline staff like to know what the arrangement of pucks is, instead of it being effectively being shown as empty